src/main/riddl/Events/eventContext.riddl

```diff
@@ -0,0 +1,197 @@
+                on command DelayEvent{ ??? // is this allowed or return error?
+                }
+                on command StartEvent{
+                  then error "Event is already started. Cannot start an event that is already in progress"
+                }
+                on command EndEvent{
+									then "morph Event entity to EventStatus.Past"
+                  and "set entity lastModifiedBy to Event.endingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command AddLiveUpdate{???}
+            }
+            //score, period, etc...
+        }
+        state Delayed is {
+            fields {
+                eventId: EventId,
+                eventInfo: EventInfo,
+                reason: String,
+                eventMetaInfo: EventMetaInfo
+                //finalScore
+            }
+            handler DelayedEventHandler is {
+                on command ScheduleEvent{
+									then "morph Event entity to EventStatus.InProgress"
+                  and "set entity lastModifiedBy to Event.delayingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command ChangeEventInfo{
+                  then "set entity eventInfo to message info"
+                  and "set entity lastModifiedBy to Event.cancellingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command CancelEvent{
+									then "morph Event entity to EventStatus.Cancelled"
+                  and "set entity lastModifiedBy to Event.cancellingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command RescheduleEvent{
+                  then "set entity evenInfo.expectedStart to Event.start"
+                  and "set entity evenInfo.expectedEnd to Event.end"
+                  and "set entity lastModifiedBy to Event.reschedulingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command DelayEvent{
+                  then "morph Event entity to EventStatus.Delayed"
+                  and "set entity eventInfo.expectedStart to eventInfo.expectedStart + Event.duration"
+                  and "set entity eventInfo.expectedEnd to eventInfo.expectedEnd + Event.duration"
+                  and "set entity lastModifiedBy to Event.delayingMember"
+                  and "set entity lastModifiedOn to today()"
+                  // what about Event.reason? Do we want to store it somewhere?
+                }
+                on command StartEvent{
+                  then "morph Event entity to EventStatus.InProgress"
+                  and "set entity lastModifiedBy to Event.startingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command EndEvent{
+									then error "End has not started. Cannot end an event that has not started."
```

Comment:
minor typo. "Only events that are InProgress may be ended. If you wish to end this event it must be started first. If you wish to cancel the event, please use the CancelEvent Command."